### ISSUE TITLE
Super-Resolution using Convolutional Neural Networks without Any Checkerboard Artifacts

### ISSUE THREAD

**Issue by mrgloom** [jantic/DeOldify#155]
Here is `CustomPixelShuffle_ICNR`
https://github.com/jantic/DeOldify/blob/554c23a2327ecfa705ca32c48ae9358ec5c96848/deoldify/unet.py#L19
As I understand it doing conv->relu->pixelshuffle->pad->blur

Which part is refer to H0 in paper? Which method (A,B,C) is used?
https://arxiv.org/pdf/1806.02658.pdf

Why not use total variation loss instead of using 'blur' using `nn.AvgPool2d`?

How crucial is weight initialization, i.e. `icnr` init and `weight_norm`?

**Comment by jantic** [comment]
I can't really answer most of these questions unless I really dig.  This particular code actually comes straight from the fastai library with only minor modification (mine isn't using weight norm though now I look at it the comments still remain that say it does).  So I guess the weight_norm isn't super crucial because I certainly don't see artifacts from using batch norm.  But beyond that I can't tell you what exactly motivated the decisions on this.  For me it "just works" (sorry if that's not a great answer but it's honest!).
